### PR TITLE
Added multiple language test using maven verifier 

### LINKED ISSUE #12: Support path pattern parameter
### Situation

OFT looks for the tags inside the [supported file extensions](https://github.com/itsallcode/openfasttrace/blob/develop/importer/tag/src/main/java/org/itsallcode/openfasttrace/importer/tag/TagImporterFactory.java#L42-L62). Usually, for the non Java sources with these extension, you could use [`build-helper-maven-plugin`](https://www.mojohaus.org/build-helper-maven-plugin/usage.html) to add additional sources. For example, Lua sources.

For other file extension which are not in the supported list, OFT provides pattern (both regex, glob) API. It would be helpful if this plugin also supports it.

### Expected Solution

For example, for Scala sources src/main/scala, I would like to define a `pathPattern` as below:

```xml
<plugin>
    <groupId>org.itsallcode</groupId>
    <artifactId>openfasttrace-maven-plugin</artifactId>
    <version>1.0.1</version>
    <executions>
        <execution>
            <id>trace-requirements</id>
            <goals>
                <goal>trace</goal>
            </goals>
        </execution>
    </executions>
    <configuration>
        <reportOutputFormat>html</reportOutputFormat>
        <reportVerbosity>ALL</reportVerbosity>
        <reportShowOrigin>true</reportShowOrigin>
        <pathPattern>**.scala</pathPattern>
    </configuration>
</plugin>
```

`pathPattern` is initial suggestion, we could change it to `pathConfigPattern` or other nice descriptive parameter name.

### Alternative Solution

Another solution would be to add user additional file extensions to the supported list and do release of oft and oft-maven plugins.

### PR DESCRIPTION
**Pull request by morazow**
Fixes #12.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -35,5 +35,5 @@ jobs:
       uses: scacap/action-surefire-report@v1
       if: always()
       with:
-        report_paths: '**/target/surefire-reports/TEST-*.xml'
+        report_paths: '**/target/surefire-reports/TEST-*.xml,**/target/failsafe-reports/TEST-*.xml'
```
Comment (kaklakariada) [inline .github/workflows/build.yml:38]: 👍

File: src/test/java/org/itsallcode/openfasttrace/maven/TraceMojoVerifierTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        final MavenIntegrationTestEnvironment mvnITEnv = new MavenIntegrationTestEnvironment();
+        mvnITEnv.installPlugin(CURRENT_PLUGIN_JAR, CURRENT_PLUGIN_POM);
+    }
+
+    @Test
+    public void testTracingWithPathPatternSuccessful() throws Exception
+    {
+
+        final Verifier verifier = new Verifier(PROJECT_WITH_PATH_PATTERN.toAbsolutePath().toString());
+        verifier.executeGoal("verify");
```
Comment (kaklakariada) [inline src/test/java/org/itsallcode/openfasttrace/maven/TraceMojoVerifierTest.java:59]: 👍

File: src/test/java/org/itsallcode/openfasttrace/maven/TraceMojoVerifierTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        final Verifier verifier = new Verifier(PROJECT_WITHOUT_PATH_PATTERN.toAbsolutePath().toString());
+        try
+        {
+            verifier.executeGoal("verify");
+        }
+        catch (final Exception exception)
+        {
+            verifier.verifyTextInLog("Tracing found 1 out of 4 items");
+            throw exception;
+        }
```
Comment (kaklakariada) [inline src/test/java/org/itsallcode/openfasttrace/maven/TraceMojoVerifierTest.java:77]: This is not nice. I created #21 to upgrade to JUnit 5 and refactor this test. For now it's ok.

File: src/test/resources/project-with-path-pattern/src/main/java/source.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * #L%
+ */
+// [impl->dsn~feature-one~1]
```
Comment (kaklakariada) [inline src/test/resources/project-with-path-pattern/src/main/java/source.java:22]: Does this compile? Don't we need a class definition?

File: src/test/resources/project-with-path-pattern/src/main/java/source.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * #L%
+ */
```
Comment (kaklakariada) [inline src/test/resources/project-with-path-pattern/src/main/java/source.java:21]: The license header distracts from the real intention of this file. Please add an exclusion to the license checker plugin and remove the header.

**Review by kaklakariada — CHANGES_REQUESTED** [review]
I am sorry, but this won't work. `PathConfig` is not used to add input files. Instead it is used to configure the [TagImporter](https://github.com/itsallcode/openfasttrace/blob/develop/importer/tag/src/main/java/org/itsallcode/openfasttrace/importer/tag/TagImporter.java#L64) to allow using [short tags](https://github.com/itsallcode/openfasttrace/blob/develop/importer/tag/src/main/java/org/itsallcode/openfasttrace/importer/tag/ShortTagImportingLineConsumer.java#L35) that don't require specifying the artifact type. Instead you only write `[[item-id:revision]]`.

To add the feature you would like to have, we need to extend oft itself.

**Comment by morazow** [comment]
Hey @kaklakariada,

Thanks for the review! 

I had to pull some testing libraries (verifier, exasol maven testing) to get tests to work since needed to run tracing after verify. There is still issue with installing current maven plugin locally for verifier.

> To add the feature you would like to have, we need to extend oft itself.

Okay, then I am going to close this pull request. I did not get the idea for short tags. 

To move forward, should I add some more language extensions to the oft core? Then we can release oft and oft-maven-plugin.

**Comment by kaklakariada** [comment]
I think adding the testing libraries is a good idea and we should keep it. If you want you can adapt this PR accordingly.

Adding the language extensions to the core is the quickest and easiest solution, although not the nicest.

**Comment by morazow** [comment]
Hey @kaklakariada,

> Adding the language extensions to the core is the quickest and easiest solution, although not the nicest.

I also initially had same idea, then to make it more general followed with this issue.

Okay, then let's do as following: I am going to finish this PR as integration test using testing dependencies and address your suggestions. At the same time open another PR for more language extensions in the core.

**Comment by morazow** [comment]
Then we can get a new release with included patches soon.

File: pom.xml [hunk 6]
```diff
@@ -253,6 +297,7 @@
                     <dryRun>false</dryRun>
                     <excludes>
                         <exclude>plugin/org/itsallcode/openfasttrace/maven/HelpMojo.java</exclude>
+                        <exclude>test/resources/*/src/**</exclude>
```
Comment (morazow) [inline pom.xml:300]: I have added this, but header is available in some of previous tests. Please let me know if I should remove from them also.

File: src/test/resources/project-with-path-pattern/src/main/java/source.java (hunk 4 above)
Comment (morazow) [reply src/test/resources/project-with-path-pattern/src/main/java/source.java]: I did understand yet how verifier plugin runs the test, but it did not complain, yes it works.

File: src/test/resources/project-with-path-pattern/src/main/java/source.java (hunk 5 above)
Comment (morazow) [reply src/test/resources/project-with-path-pattern/src/main/java/source.java]: Done.

**Comment by morazow** [comment]
Hello @kaklakariada,

Please have a look! 

I have updated the pr and renamed the title.

File: pom.xml (hunk 6 above)
Comment (kaklakariada) [reply pom.xml:300]: Thank you, that's fine. We can do this later!

**Review by kaklakariada — APPROVED** [review]
Thank you for the changes!

**Pull request merged**